### PR TITLE
Workflow to publish a release

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Publish `@dfinty/verifiable-credentials` to NPM.

# Changes

* Add the Github workflow that will publish to NPM new releases created in Github.

# Tests

This is the same workflow used in ic-js, hardware-wallete-cli and generative-logo repository.

### CODE REVIEW

**Review by frederikrothenberger — APPROVED** [review]
LGTM, thanks!

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    runs-on: ubuntu-20.04
+    env:
+      working-directory: js-library
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "18.x"
```
Comment (peterpeterparker) [inline .github/workflows/publish.yml:18]: Please use Node LTS (20+)

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          registry-url: "https://registry.npmjs.org"
+      - run: npm install -g npm
```
Comment (peterpeterparker) [inline .github/workflows/publish.yml:20]: Not sure you need this step actually

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        run: npm run build
+      - name: Test
+        working-directory: ${{env.working-directory}}
+        run: npm test
+      - name: Set up npm
+        working-directory: ${{env.working-directory}}
+        run: printf '%s\n' '//registry.npmjs.org/:_authToken=${NODE_AUTH_TOKEN}' registry=https://registry.npmjs.org/ always-auth=true >> .npmrc
+      - name: Publish
+        working-directory: ${{env.working-directory}}
+        run: npm publish --access public
```
Comment (peterpeterparker) [inline .github/workflows/publish.yml:35]: Provenance is missing

Comment (lmuntaner) [reply .github/workflows/publish.yml]: Thanks!

File: .github/workflows/publish.yml (hunk 1 above)
Comment (lmuntaner) [reply .github/workflows/publish.yml]: Done!

**Comment by peterpeterparker** [comment]
Thanks for integrating the feedback! Sorry for their conciseness, I was reviewing it quickly.

**Pull request merged**